appcues/src/main/java/com/appcues/statemachine/Action.kt

```diff
@@ -1,16 +1,19 @@
 package com.appcues.statemachine
 
 import com.appcues.data.model.Experience
+import com.appcues.data.model.StepReference
 
 internal sealed class Action {
     data class StartExperience(val experience: Experience) : Action()
-    data class StartStep(val stepReference: StepReference) : Action()
-    object RenderStep : Action()
+    data class MoveToStep(val stepReference: StepReference) : Action()
+    data class StartStep(val nextFlatStepIndex: Int, val nextStepContainerIndex: Int) : Action()
+    data class RenderStep(val metadata: Map<String, Any?>) : Action()
```

Comment:
RenderStep has the produced metadata, this gives us better control of the states, instead of doing an elaborate work during BeginningStepState we can get to Beginning and invoke PresentationEffect, which will do all of this for us and return RenderStep(metadata), isolating the produceMetada, PresentingTrait.present() and retries